cli/src/main/java/org/jboss/sbomer/cli/feature/sbom/processor/DefaultProcessor.java

```diff
@@ -266,9 +275,102 @@ public Bom process(Bom bom) {
+
         return bom;
     }
 
+    private void addMissingNpmDependencies(Bom bom) {
+        if (bom.getMetadata() == null || bom.getMetadata().getComponent() == null) {
+            // No main component
+            return;
+        }
+        Component component = bom.getMetadata().getComponent();
+        List<ExternalReference> externalReferences = getExternalReferences(component, ExternalReference.Type.BUILD_SYSTEM, SBOM_RED_HAT_PNC_BUILD_ID);
+        if (externalReferences.isEmpty()) {
+            // Not a PNC build
+            return;
+        }
+        if (externalReferences.size() > 1) {
+            log.warn("Component {} hase more then one {}/{} external reference", component.getPurl(), ExternalReference.Type.BUILD_SYSTEM, SBOM_RED_HAT_PNC_BUILD_ID);
+            return;
+        }
+        String buildID = parseBuildIdFromURL(externalReferences.get(0).getUrl());
+        if (buildID == null) {
+            log.warn("Could not parse PNC build ID from url {}", externalReferences.get(0).getUrl());
+            return;
+        }
+        Build build = pncService.getBuild(buildID);
+        if (build == null) {
+            log.warn("Could not obtain PNC build for build id {}", buildID);
+            return;
+        }
+        if (build.getBuildConfigRevision().getBuildType() == BuildType.NPM) {
+            // Not processing pure NPM builds, cyclonedx-nodejs plugin handles them correctly
+            return;
+        }
+        Collection<Artifact> npmDependencies = pncService.getNPMDependencies(buildID);
+        if (npmDependencies.isEmpty()) {
+            // No NPM dependencies to add
+            return;
+        }
+
+        addMissingNpmDependencies(bom, npmDependencies, build.getBuildConfigRevision().getBuildType());
+    }
+
+    private void addMissingNpmDependencies(Bom bom, Collection<Artifact> npmDependencies, BuildType buildType) {
+        Set<String> listedPurls = bom.getComponents().stream()
+            .map(DefaultProcessor::getPackageURL)
+            .map(PackageURL::getCoordinates)
+            .collect(Collectors.toSet());
+
+        Set<String> dependencyRefs = new HashSet<>();
+        for (Artifact artifact : npmDependencies) {
+            String coordinates;
+            try {
+                coordinates = new PackageURL(artifact.getPurl()).getCoordinates();
+            } catch (MalformedPackageURLException e) {
+                log.warn("Unable to parse artifact purl '{}'.", artifact.getPurl());
+                continue;
+            }
+            if (listedPurls.contains(coordinates)) {
+                continue;
+            }
+            Component newComponent = createComponent(artifact, Component.Scope.REQUIRED, Component.Type.LIBRARY, buildType);
+            setArtifactMetadata(newComponent, artifact, pncService.getApiUrl());
+            setPncBuildMetadata(newComponent, artifact.getBuild(), pncService.getApiUrl());
+            bom.addComponent(newComponent);
```

Comment:
@vibe13 I think the component is added here. And it looks that the tests cover this.